lib/source.c

```diff
@@ -109,6 +116,12 @@ tlog_source_read(struct tlog_source *source, struct tlog_pkt *pkt)
 
     assert(grc == TLOG_RC_OK || tlog_pkt_is_void(pkt));
     assert(tlog_source_is_valid(source));
+#ifndef NDEBUG
+    tlog_timespec_sub(&pkt->timestamp, &source->last_timestamp, &diff);
+    assert(tlog_timespec_cmp(&diff, &tlog_timespec_zero) >= 0);
+    assert(tlog_timespec_cmp(&diff, &tlog_delay_max_timespec) <= 0);
+    source->last_timestamp = pkt->timestamp;
```

Comment:
One last thing: this needs to be checked only when `pkt` is not "void". Otherwise `pkt->timestamp` can be invalid.

Reply:
I see the `tlog_pkt_is_void(pkt)` already check before `ifndef`. It means `pkt->timestamp` will be valid